### PR TITLE
Update to use v2.5.6 of Firmata library

### PR DESCRIPTION
**Pull request by sidwarkd**
Tested StandardFirmata.ino example in all 3 development environments targeting a Particle Photon.

### CODE REVIEW

**Comment by soundanalogous** [comment]
There was a bug in v2.5.5 that broke Firmata strings and I2C. You should update to v2.5.6. Here's the change: https://github.com/firmata/arduino/commit/f18eb7519246a78563c3cae638cb11d01568fcbb.

File: src/Boards.h [hunk 1]
```diff
@@ -123,7 +123,7 @@ writePort(port, value, bitmask):  Write an 8 bit port.
 // #define TOTAL_PINS              21 //defined in spark_wiring.h
 // #define TOTAL_ANALOG_PINS       8 //defined in spark_wiring.h
 #define VERSION_BLINK_PIN       7
-#define IS_PIN_DIGITAL(p)       ((p) >= 0 && (p) < TOTAL_PINS && (p) != 8 && (p) != 9  && (p) != 20)
+#define IS_PIN_DIGITAL(p)       ((p) >= 0 && (p) < TOTAL_PINS && (p) != 8 && (p) != 9  && (p) != 20 && (p) != 21 && (p) != 22 && (p) != 23)
 #define IS_PIN_ANALOG(p)        ((p) >= FIRST_ANALOG_PIN && (p) < (FIRST_ANALOG_PIN+TOTAL_ANALOG_PINS))
 #define IS_PIN_PWM(p)           ((p) == 0 || (p) == 1 || (p) == 10 || (p) == 11 || (p) == 14 || (p) == 15 || (p) == 16 || (p) == 17)
```
Comment (soundanalogous) [inline src/Boards.h]: You may be able to use the macro [digitalPinHasPWM(p)](https://github.com/spark/firmware/blob/develop/wiring/inc/spark_wiring_arduino.h#L125) here.

File: src/Boards.h [hunk 2]
```diff
@@ -123,7 +123,7 @@ writePort(port, value, bitmask):  Write an 8 bit port.
 // #define TOTAL_PINS              21 //defined in spark_wiring.h
 // #define TOTAL_ANALOG_PINS       8 //defined in spark_wiring.h
 #define VERSION_BLINK_PIN       7
```
Comment (soundanalogous) [inline src/Boards.h]: change 7 to `LED_BUILTIN` per https://github.com/spark/firmware/blob/develop/wiring/inc/spark_wiring_arduino.h#L160

File: src/FirmataMarshaller.h [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  #include <cstdint>
+#else
+  #include <stddef.h>
+  #include <stdint.h>
+#endif
+
+#ifndef PARTICLE
+#include <Stream.h>
+#else
+#include "Particle.h"
```
Comment (soundanalogous) [inline src/FirmataMarshaller.h]: I don't think this is necessary anymore now that Particle exposes Stream.h in the user space. However if you've tried and encountered errors or need to maintain backwards compatibility then including Particle.h is still necessary.

**Comment by sidwarkd** [comment]
@soundanalogous I incorporated the 2.5.6 changes as well as the other items of feedback you gave.

**Comment by soundanalogous** [comment]
Looks good, thanks!

Once I get my Photon board later this week I can look into additional improvements such as getting StandardFirmataWiFi to work.

**Pull request merged**